app/src/main/java/no/usn/mob3000/ui/screens/SettingsScreen.kt

```diff
@@ -39,23 +47,72 @@ import no.usn.mob3000.ui.theme.DefaultButton
 @OptIn(ExperimentalMaterial3Api::class)
 @Composable
 fun SettingsScreen(
+    logoutState: Flow<LogoutState>,
+    logoutStateReset: () -> Unit,
     onLogoutClick: () -> Unit,
     onLoginClick: () -> Unit,
+    accountDeleteState: Flow<DeleteAccountState>,
+    accountDeleteStateReset: () -> Unit,
+    onDeleteAccountClick: () -> Unit,
     onAdminDashboardClick: () -> Unit,
     selectedTheme: String,
     selectedLanguage: String,
     onThemeChange: (String) -> Unit,
     onLanguageChange: (String) -> Unit
 ) {
+    val stateLogout by logoutState.collectAsState(initial = LogoutState.Idle)
+    var showLogoutConfirmation by remember { mutableStateOf(false) }
+    val stateAccountDeletion by accountDeleteState.collectAsState(initial = DeleteAccountState.Idle)
+    var showAccountDeletionConfirmation by remember { mutableStateOf(false) }
+
     var themeExpanded by remember { mutableStateOf(false) }
     var languageExpanded by remember { mutableStateOf(false) }
 
+    /**
+     * This is sort of viscerally disgusting, but also somewhat necessary.
+     *
+     * Without this, any successes and error dialogues will be shown indefinitely,
+     * which *can* be handled with a timer or some such, but honestly? This is just
+     * much easier, and works practically just as well for less complexity.
+     *
+     * So as much as this disgusts me, it's here to stay.
```

Comment:
Totally reliable code. As long as it works like a charm 👯

Reply:
🗿